src/NuGet.Clients/PackageManagement.UI/Actions/NuGetProjectUpgrader.cs

```diff
@@ -0,0 +1,190 @@
+
+            // TODO: How should we handle a failure in uninstalling a package (unfortunately ExecuteNuGetProjectActionsAsync()
+            // doesn't give us any useful information about the failure).
+            await context.PackageManager.ExecuteNuGetProjectActionsAsync(nuGetProject, actions, uiService.ProgressWindow, CancellationToken.None);
+
+            // If there were packages we didn't uninstall because we couldn't find them, they will still be present in
+            // packages.config, so we'll have to delete that file now.
+            if (File.Exists(packagesConfigFullPath))
+            {
+                FileSystemUtility.DeleteFile(packagesConfigFullPath, uiService.ProgressWindow);
+                msBuildNuGetProjectSystem.RemoveFile(Path.GetFileName(packagesConfigFullPath));
+            }
+
+            // 3. Create stub project.json file
+            progressData = new ProgressDialogData(Resources.NuGetUpgrade_WaitMessage, Resources.NuGetUpgrade_Progress_CreatingProjectJson);
+            progress.Report(progressData);
+
+            var json = new JObject();
+
+            // Target framework
+            var targetNuGetFramework = msBuildNuGetProjectSystem.TargetFramework;
+            json["frameworks"] = new JObject {[targetNuGetFramework.GetShortFolderName()] = new JObject()};
+
+            // Runtimes
+            var runtimeStub = msBuildNuGetProjectSystem.GetPropertyValue("TargetPlatformIdentifier") == "UAP"
+                ? "win10"
+                : "win";
+            var runtimes = new JObject();
+            var supportedPlatforms = msBuildNuGetProjectSystem.SupportedPlatforms;
+
+            if (supportedPlatforms.Any())
+            {
+                foreach (var supportedPlatformString in supportedPlatforms)
+                {
+                    if (string.IsNullOrEmpty(supportedPlatformString) || supportedPlatformString == "Any CPU")
+                    {
+                        runtimes[runtimeStub] = new JObject();
+                    }
+                    else
+                    {
+                        runtimes[runtimeStub + "-" + supportedPlatformString.ToLowerInvariant()] = new JObject();
+                    }
+                }
+            }
+            else
+            {
+                runtimes[runtimeStub] = new JObject();
+            }
+
+            json["runtimes"] = runtimes;
+
+            // Write out project.json and add it to the project
+            var projectJsonFileName = Path.Combine(msBuildNuGetProjectSystem.ProjectFullPath, PackageSpec.PackageSpecFileName);
+            WriteProjectJson(projectJsonFileName, json);
+            msBuildNuGetProjectSystem.AddExistingFile(PackageSpec.PackageSpecFileName);
+
+            // Reload the project, and get a reference to the reloaded project
+            var uniqueName = msBuildNuGetProjectSystem.ProjectUniqueName;
+            solutionManager.SaveProject(nuGetProject);
```

Comment:
I think we're good here. Internal API, and I can't see any scenario where these can be null.